### PR TITLE
@dzucconi -> fallback partnershow cover image to artwork image

### PR DESCRIPTION
**Pull request by 1aurabrown**
see the discussion in here about show cover images: https://github.com/artsy/force/issues/4768

lmk if this is not the way to do this.

### CODE REVIEW

File: schema/partner_show.js [hunk 1]
```diff
@@ -105,7 +106,16 @@ const PartnerShowType = new GraphQLObjectType({
     },
     cover_image: {
       type: Image.type,
-      resolve: ({ image_versions, image_url }) => Image.resolve({ image_versions, image_url }),
+      resolve: ({ id, partner, image_versions, image_url }) => {
+        if (image_versions && image_versions.length && image_url) {
+          return Image.resolve({ image_versions, image_url });
+        }
```
Comment (dzucconi) [inline schema/partner_show.js]: This should just be handled by the [image normalization](https://github.com/artsy/metaphysics/blob/master/schema/image/normalize.js) which is [https://github.com/artsy/metaphysics/blob/a68f56e4f39108957d5ec66b8832f49caaa26fa7/schema/image/index.js#L77](what that resolve is anyway)

Comment (dzucconi) [inline schema/partner_show.js]: OH. Sorry I see now what's going on. Ignore

File: schema/partner_show.js [hunk 2]
```diff
@@ -105,7 +106,16 @@ const PartnerShowType = new GraphQLObjectType({
...
+        return gravity(`partner/${partner.id}/show/${id}/artworks`, { size: 1, published: true })
+          .then(artworks => {
+            const { images } = artworks[0];
+            return Image.resolve(_.find(images, { is_default: true }) || _.first(images));
```
Comment (dzucconi) [inline schema/partner_show.js]: We should expose a helper function for pulling out default images since I see this happening in a few places.

**Comment by dzucconi** [comment]
Works for me.

File: schema/partner_show.js (hunk 2 above)
Comment (1aurabrown) [inline schema/partner_show.js]: ya

**Comment by mzikherman** [comment]
I don't think this is the right solution.

This overrides the cover image with a fallback...._which might not be shareable_.  But how is the client supposed to know if the cover in this case can be shared, or is one of these fallbacks?

Now, this show page: https://www.artsy.net/show/museum-boijmans-van-beuningen-ugo-rondinone shows this image when you share on Facebook:

screen shot 2016-04-19 at 11 28 52 am
 
That's precisely an image of an artwork that the partner decided to not allow to be shared on social media. By putting the fallback in the same field as `cover_image`, we lose that distinction.

I think we should revert this and either:

a) Add it into a _new_ field and not override `cover_image` (as discussed in the previous issue)
b) Confirm with the PM's in #partner-success that the social media share stuff doesn't matter at all, in which case we can propagate this fallback up to Gravity.

**Comment by dzucconi** [comment]
We could also just `null` it when the shareable status comes back false?

**Comment by mzikherman** [comment]
I don't quite follow the `null` thing?

Basically, I don't think we can place this fallback into the existing `cover_image` field, b/c of the shareable distinction. So I think a new field makes sense.

**Comment by mzikherman** [comment]
Another alternative - keep this as-is but update the show page to only render the `og:image` meta tag if the cover is sharable. That would mean somehow enhancing the response w/ that piece of info.

I guess that's a third option:

a) introduce new field `cover_image_not_for_sharing` and use that
b) Keep this as-is, but add a field that indicates whether it can be shared, and update the show page sharing to respect that
c) Follow up with #partner-success on whether we can drop this altogether.

@1aurabrown @dzucconi thoughts?

**Comment by dzucconi** [comment]
Something we can do is make the meta tag the special case by writing it's own meta schema (which would include/or not a shareable image):
https://github.com/artsy/metaphysics/blob/master/schema/artwork/meta.js

The show pages are already using metaphysics so it should be pretty simple to do that.

**Comment by mzikherman** [comment]
ooo good idea. I am down for giving that a shot, I can do it since I've touched the show page last.

Ok so this is cool as-is for now, and I'll migrate the show page meta stuff up here too (https://trello.com/c/aran7nBi/286-shift-meta-tag-stuff-to-metaphysics-for-the-show-page-re-shareable-covers).

Sweet, I like that! Sorry for all the back n forth @1aurabrown

**Comment by 1aurabrown** [comment]
:( I agree there should be some separation between the image that will be included in the meta tag, and places where we want the UI to include some sort of fallback image when the show has no cover image. It seems like an accidental mixing of concerns to have the 'current show' widget not show a fallback image because the schema that powers that data would then also affect the meta tags on a show page.

**Comment by mzikherman** [comment]
> :( I agree there should be some separation between the image that will be included in the meta tag, and places where we want the UI to include some sort of fallback image when the show has no cover image. It seems like an accidental mixing of concerns to have the 'current show' widget not show a fallback image because the schema that powers that data would then also affect the meta tags on a show page.

Exactly! **puts on robe and wizard hat** "By the power of GraphQL, I orchestrate thee!" **shakes staff**

**Comment by oxaudo** [comment]
So when we display shows in cms we display the cover image that is set here: https://github.com/artsy/gravity/blob/master/app/models/domain/partner_show.rb#L169

Seems like that is not valid (not what is going to be displayed on artsy) in some cases?

**Pull request merged**